src/NUnitFramework/framework/Constraints/StringConstraint.cs

```diff
@@ -92,7 +140,7 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
             var stringValue = ConstraintUtils.RequireActual<string>(actual, nameof(actual), allowNull: true);
 
-            return new ConstraintResult(this, actual, Matches(stringValue));
+            return new ConstraintResult(this, actual, cultureInfo is not null ? Matches(stringValue, cultureInfo) : Matches(stringValue));
```

Comment:
What about when _comparisonType_ is selected.
Suggest to add an `Match` overload which also expects `StringComparison`